daliuge-translator/dlg/dropmake/pg_generator.py

```diff
@@ -250,7 +250,7 @@ def resource_map(pgt, nodes, num_islands=1, co_host_dim=True):
     if type(pgt[0]) is str:
         pgt = pgt[1]  # remove the graph name TODO: we may want to retain that
     for drop_spec in pgt:
-        if drop_spec != {}:
+        if drop_spec != {} and "node" in drop_spec:
```

Comment:
I would agree with the `sourcery` review here; it's more Pythonic to write: 

```python
if drop_spec and "node" in drop_spec:
```

Reply:
We actually don't need the first part at all, since "node" in drop_spec covers both cases correctly anyway. However the more serious issue with this if statement is that we assume that the "island" key is there if the "node" key is there, which is not necessarily the case. It also hard-codes the island port which is wrong as well and I have no clue whatsoever, why it is replacing the port if it is given with that default port in the first place. Thus this seems to be a bigger fish...

Reply:
But I've just committed a change addressing those issues as well. Seems to be fine...